game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/CombatValue.java

```diff
@@ -0,0 +1,98 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.TerritoryEffect;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.Matches;
+import java.util.Collection;
+
+public interface CombatValue {
+
+  StrengthAndRollCalculator getRoll();
+
+  StrengthAndRollCalculator getStrength();
+
+  boolean isDefending();
+
+  GameData getGameData();
+
+  static CombatValue buildMain(
+      final Collection<Unit> allEnemyUnitsAliveOrWaitingToDie,
+      final Collection<Unit> allFriendlyUnitsAliveOrWaitingToDie,
+      final boolean defending,
```

Comment:
boolean parameters are often best avoided. I suggest moving these factor methods to a factory class. In that case you can have an internal factory method (private) that has a boolean paramter, but the public methods would be `buildMainOffensiveCombatValue` and `buildMainDefensiveCombatValue` and `buildAaDefensiveCombatValue`

If those methods are in a `CombatValueFactory`, the suffix probably could be dropped, ie:
`buildMainOffensive`, etc..

WDYT?

Reply:
I'd love to replace all the usage of the boolean `defending` with the BattleState.Side enums.  It is used in a lot of places.  If you are ok with that, I could do that as a separate PR.

As for the factory change, let's wait on that.  I have a branch dependent on this where the construction of the `CombatValue` objects is moved up the call tree.  I think that might be a better PR to discuss how they are constructed.  It might make it more obvious on their construction.